src/Shared/SecretsStore.cs

```diff
@@ -12,83 +11,92 @@ namespace Microsoft.Extensions.SecretManager.Tools.Internal;
 /// <summary>
 /// Adapted from dotnet user-secrets at https://github.com/dotnet/aspnetcore/blob/482730a4c773ee4b3ae9525186d10999c89b556d/src/Tools/dotnet-user-secrets/src/Internal/SecretsStore.cs
 /// </summary>
-internal sealed class SecretsStore
+internal abstract class KeyValueStore
```

Comment:
I'm not sure we need this abstraction. SecretsStore is just about reading/writing the `secrets.json` file. The new Store should be about reading/writing files to the app directory (wherever it is that we decide).